### ISSUE TITLE
Why gradients don't seem to pass through Embedding ?

### ISSUE THREAD

**Issue by LuCeHe** [keras-team/keras#12270]
I'm not sure if the gradients are passing through the embedding layer properly:

``` python

import tensorflow as tf

import keras.backend as K
from keras.models import Sequential, Model
from keras.layers import Dense, concatenate, Input, Conv2D, Embedding, \
                         Bidirectional, LSTM, Lambda, TimeDistributed, \
                         RepeatVector, Activation


vocabSize = 3
embDim = 5
rnn = LSTM(vocabSize, return_sequences=True)
embedding = Embedding(vocabSize, embDim)
      
      
input_questions = Input(shape=(None,), name='question')
embed = embedding(input_questions)
rnn_output = rnn(embed)    
softmax = TimeDistributed(Activation('softmax'))(rnn_output)

# these gradients don't work, FIXME!
grad = tf.gradients(xs=input_questions, ys=rnn_output)
print('grad (xs=input_questions, ys=rnn_output):   ', grad)            
grad = tf.gradients(xs=input_questions, ys=embed)
print('grad (xs=input_questions, ys=embed):        ', grad)            
grad = tf.gradients(xs=input_questions, ys=softmax)
print('grad (xs=input_questions, ys=softmax):      ', grad)    
grad = tf.gradients(xs=embed, ys=softmax)
print('grad (xs=embed, ys=softmax):                ', grad)    

model = Model(inputs=input_questions, outputs=softmax)

```

The output of this code is

```python
grad (xs=input_questions, ys=rnn_output):    [None]
grad (xs=input_questions, ys=embed):         [None]
grad (xs=input_questions, ys=softmax):       [None]
grad (xs=embed, ys=softmax):                 [<tf.Tensor 'gradients_10/lstm_3/transpose_grad/transpose:0' shape=(?, ?, 5) dtype=float32>]
```

**Comment by LuCeHe** [comment]
any news @gabrieldemarmiesse ??

**Comment by Dref360** [comment]
The embedding layer is implemented as K.gather which is not differentiable, so there is no gradient.

**Comment by gabrieldemarmiesse** [comment]
Thanks @Dref360 for looking into this. Closing the issue as this is the expected behaviour of the embedding layer.

**Comment by edervishaj** [comment]
> The embedding layer is implemented as K.gather which is not differentiable, so there is no gradient.

In the `build` method of the layer a new trainable weight is added by `add_weight` with `trainable=True` by default:
```python3
def build(self, input_shape):
        self.embeddings = self.add_weight(
            shape=(self.input_dim, self.output_dim),
            initializer=self.embeddings_initializer,
            name='embeddings',
            regularizer=self.embeddings_regularizer,
            constraint=self.embeddings_constraint,
            dtype=self.dtype)
        self.built = True
```
Doesn't this make the weights of Embedding layer trainable?
